src/cmd/ksh93/tests/special-dev-paths.sh

```diff
@@ -0,0 +1,159 @@
+#
+# Verify that the special /dev paths such as /dev/stdin, /dev/fd/0, and /dev/tcp/host/port work when
+# used in a test and actually opened. This is primarily meant to exercise the `sh_open()` functions
+# handling of special paths.
+#
+readonly host=localhost
+readonly port=65123
+
+# Some tests will want to verify whether or not a fd was allocated. We could use the `lsof` command
+# but don't want to add that as a dependency. Instead just do the verification on platforms which
+# support /proc/self/fd.
+[[ -d /proc/self/fd ]] && readonly fd_check=yes || readonly fd_check=no
```

Comment:
Probably it would be bettter to just declare variables like `fd_check` with default value of `no` at top and then change it to `yes` if this test succeeds.

Reply:
Agreed. I was just trying a different idiom to see how it looked/worked.